core/runtime/binaryen/runtime_manager.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_RUNTIME_BINARYEN_RUNTIME_API_RUNTIME_MANAGER
+#define KAGOME_CORE_RUNTIME_BINARYEN_RUNTIME_API_RUNTIME_MANAGER
+
+#include <binaryen/wasm-interpreter.h>
+
+#include "common/blob.hpp"
+#include "common/logger.hpp"
+#include "crypto/hasher.hpp"
+#include "extensions/extension_factory.hpp"
+#include "outcome/outcome.hpp"
+#include "runtime/wasm_provider.hpp"
+#include "runtime_external_interface.hpp"
+
+namespace kagome::runtime::binaryen {
+
+  class RuntimeManager {
+   public:
+    enum class Error { EMPTY_STATE_CODE = 1, INVALID_STATE_CODE };
+
+    RuntimeManager(
+        std::shared_ptr<runtime::WasmProvider> wasm_provider,
+        std::shared_ptr<extensions::ExtensionFactory> extension_factory,
+        std::shared_ptr<crypto::Hasher> hasher)
+        : wasm_provider_(std::move(wasm_provider)),
+          extension_factory_(std::move(extension_factory)),
+          hasher_(std::move(hasher)) {
+      BOOST_ASSERT(wasm_provider_);
+      BOOST_ASSERT(extension_factory_);
```

Comment:
I think hasher should be validated either.

Reply:
Sure